packages/webauthn/index.ts

```diff
@@ -0,0 +1,196 @@
+    publicKeyY: credPubKeyYUint256
+  } as WebauthnRegistration
+}
+
+export const requestWebauthn = async (
+  params: WebauthnRequest
+): Promise<WebauthnAuthentication> => {
+  const authResJson = await startAuthentication({
+    ...(params.credentialId
+      ? {
+          allowCredentials: [
+            { id: params.credentialId, type: defaultWebauthn.pubKeyCredType }
+          ] as PublicKeyCredentialDescriptorJSON[]
+        }
+      : {}),
+    userVerification: defaultWebauthn.userVerificationRequirement,
+    challenge: params.challenge
+  } as PublicKeyCredentialRequestOptionsJSON)
+
+  const authDataUrlB64 = authResJson.response.authenticatorData
+  const authDataHex = `0x${isoUint8Array.toHex(
+    isoBase64URL.toBuffer(authDataUrlB64)
+  )}`
+  const clientDataJsonUrlB64 = authResJson.response.clientDataJSON
+  const clientDataJsonUtf8 = isoBase64URL.toString(clientDataJsonUrlB64)
+  const sigUrlB64 = authResJson.response.signature
+  const [sigRUint, sigSUint] = parseSignature(sigUrlB64)
+  //   console.log(`[Test][P256]`)
+
+  //   console.log(`[Test][P256] r: ${p256Sig.r}`)
+  //   console.log(`[Test][P256] s: ${p256Sig.s}`)
+  return {
+    authenticatorData: authDataHex,
+    clientDataJson: clientDataJsonUtf8,
+    sigantureR: sigRUint,
+    signatureS: sigSUint
+  } as WebauthnAuthentication
+}
+
+export const defaultWebauthn = {
+  attestationConveyancePreference: "none" as AttestationConveyancePreference,
+  // This Relying Party will accept either an ES256 or RS256 credential, but prefers an ES256 credential.
+  pubKeyCredAlgEs256: -7 as COSEAlgorithmIdentifier, // ES256 (Webauthn's default algorithm)
+  pubKeyCredAlgRs256: -257 as COSEAlgorithmIdentifier, // RS256 (for Windows Hello and others)
+  // Try to use UV if possible. This is also the default.
+  pubKeyCredType: "public-key" as PublicKeyCredentialType,
+  residentKeyRequirement: "required" as ResidentKeyRequirement,
+  userVerificationRequirement: "required" as UserVerificationRequirement,
+  timeout: 300000, // 5 minutes
+  // Relying Party
+  rpName: "Waallet Extension",
+  // Make excludeCredentials check backwards compatible with credentials registered with U2F
+  extensions: {
+    largeBlob: {
+      support: "preferred" // "required", "preferred"
+    }
+  },
+  requireResidentKey: true
+}
+
+const parseSignature = (signature: UrlB64String): [bigint, bigint] => {
+  if (!isoBase64URL.isBase64url(signature)) {
+    console.log(`${signature} is not Base64Url`)
+    return [BigInt(0), BigInt(0)] as const
+  }
+  const n = BigInt(
+    "0xFFFFFFFF00000000FFFFFFFFFFFFFFFFBCE6FAADA7179E84F3B9CAC2FC632551"
+  )
+  const p256Sig = p256.Signature.fromDER(
+    isoUint8Array.toHex(isoBase64URL.toBuffer(signature))
+  )
+  // If your library generates malleable signatures, such as s-values in the upper range, calculate a new s-value
+  let p256SigS = p256Sig.s
+  if (p256SigS > n / BigInt(2)) {
+    p256SigS = n - p256SigS
+  }
+  return [p256Sig.r, p256SigS]
```

Comment:
nit: No other place uses `p256Sig.s`, so we can remove `p256SigS` this temporary variable, and mutate `p256Sig.s` directly.

Reply:
It seems not to be functioning correctly here. When I attempt to replace `p256SigS` with `p256Sig.s`, I encounter the error message `Cannot assign to 's' because it is a read-only property.`